// @microsoft/dev-tunnels-ssh/progress.js
"use strict";
//
//  Copyright (c) Microsoft Corporation. All rights reserved.
//
Object.defineProperty(exports, "__esModule", { value: true });
exports.Progress = void 0;
/**
 * Connection progress
 */
var Progress;
(function (Progress) {
    // Client connection to relay
    Progress["OpeningClientConnectionToRelay"] = "OpeningClientConnectionToRelay";
    Progress["OpenedClientConnectionToRelay"] = "OpenedClientConnectionToRelay";
    // Host connection to relay
    Progress["OpeningHostConnectionToRelay"] = "OpeningHostConnectionToRelay";
    Progress["OpenedHostConnectionToRelay"] = "OpenedHostConnectionToRelay";
    // SSH Session Connection
    Progress["OpeningSshSessionConnection"] = "OpeningSshSessionConnection";
    Progress["OpenedSshSessionConnection"] = "OpenedSshSessionConnection";
    Progress["StartingProtocolVersionExchange"] = "StartingProtocolVersionExchange";
    Progress["CompletedProtocolVersionExchange"] = "CompletedProtocolVersionExchange";
    Progress["StartingKeyExchange"] = "StartingKeyExchange";
    Progress["CompletedKeyExchange"] = "CompletedKeyExchange";
    Progress["StartingSessionAuthentication"] = "StartingSessionAuthentication";
    Progress["CompletedSessionAuthentication"] = "CompletedSessionAuthentication";
})(Progress = exports.Progress || (exports.Progress = {}));
